### ISSUE TITLE
Mac borked after 21 March patch

### ISSUE THREAD

**Issue by vttale** [Azurency/CQUI_Community-Edition#314]
A 2.3 GB patch hit Mac (and Linux?) on 21 March 2018 and made the basic game unplayable.   I don't have a laundry list of problems but can identify that some really fundamental functionality is at issue.  I can't move units, and can't ranged attack.  That was enough to quit the game in frustration.

I realize that this might be a version of just updating to a slightly later version of CQUI, since this is presumably just follow-on to stuff out on Windows for a while, but it isn't clear to me what version should be used.

(As an aside to Firaxis:  why do you keep doing this to me just before I'm going on a long flight and want to play?)

**Comment by ihendriks** [comment]
There was an update for Windows as well actually.

**Comment by blaskey** [comment]
No kidding. I'm on Linux and had CQUI's fall version working just fine. I start steam yesterday, see the update starting and couldn't find a way to cancel it or start the game without finishing the download.  I should have been in offline mode.

I think Mac/Linux (if you haven't purchased R&F?) is now on 1.0.0.220 - Hotfix Patch - Post-R&F - February 13, 2018

So maybe a version of CQUI from (just) before March 8 would work? I'll probably try later. I started a game without CQUI last night and it immediately drove me crazy.

**Comment by quiznilo1** [comment]
Well, I got R&F just today and am running 1.0.0.220(160861) on Linux, and ran into an issue where I couldn't open the science tree.  I disabled all of my 3rd party mods except for CQUI and still had the bug.

The mod version I have is 1d44b5e7-753e-405b-af24-5ee634ec8a01

**Comment by sql4bucks** [comment]
Same here, Mac version, got update yesterday. Right click not working, can't choose a destination with the "Move To" unit button, etc.

**Comment by grgz** [comment]
Yes @blaskey I'm now running 1.0.0.220 on macOS, without R&F, and I installed the last CQUI commit _before_ the "Update for 1.0.0.229" commit.  It seems to work ok:

https://github.com/Azurency/CQUI_Community-Edition/tree/09e729b1a16c5f2f686e84906cfbdfc7c76dc011

I'm not sure how the mod releases are managed but it might be helpful to tag or mark a release on that commit for non-R&F users running 1.0.0.220.

**Comment by blaskey** [comment]
Yeap, @grgz I pulled down a zip of 09e729b last night and played 50 or so turns with it, with no issues. Very nice to have CQUI back! I'm tempted to set my Linux steam to offline so this doesn't happen again without me being able to schedule it.

**Comment by quiznilo1** [comment]
Confirming that https://github.com/Azurency/CQUI_Community-Edition/tree/09e729b1a16c5f2f686e84906cfbdfc7c76dc011 also works with R&F expansion on linux, everything is operational.

**Comment by grgz** [comment]
I'm still a bit confused about how R&F affects the version numbers.  Is the Mac/Linux port still at 1.0.0.220 even when R&F is installed?  And are Windows users running 1.0.0.229 both with and without R&F?

**Comment by vttale** [comment]
Also confirming.  Using that build makes it playable again.

**Comment by quiznilo1** [comment]
> I'm still a bit confused about how R&F affects the version numbers. Is the Mac/Linux port still at 1.0.0.220 even when R&F is installed? And are Windows users running 1.0.0.229 both with and without R&F?

I can confirm this with regard to Mac/Linux.

**Comment by alanhogan** [comment]
Science (especially!) and Civic progress borked with the latest Mac release of Civ, and `master` CQUI CE. can't switch Science research once it is underway, can't view Science tree. There is a new Chat widget too and I can't get it to go away. Still much, much better than playing Civ without CQUI, which makes me feel blind.

**Comment by alanhogan** [comment]
@quiznilo1 You’re saying that in 09e729b1a16c5f2f686e84906cfbdfc7c76dc011, you can use the Science tree correctly, just the trackers are busted?

**Comment by quiznilo1** [comment]
> @quiznilo1 You’re saying that in 09e729b, you can use the Science tree correctly, just the trackers are busted?

That is correct.  When I turn the trackers off, the progress circle appears over the science and social policy buttons in the upper left, but without the eureka/inspiration buffer area.

[https://i.imgur.com/ZIOBZtA.jpg](url)

**Comment by alanhogan** [comment]
@quiznilo1 Great, I tried that version and can confirm your reports. Definitely good enough to play, with the minor bugs you have described!

**Comment by Azurency** [comment]
Well... `1.0.0.220` is a version between the last two versions released on windows 🤷‍♂️ (`1.0.0.216` and `1.0.0.229`) so it'll will need its own update.
Also concerning Rise and Fall and version numbers : Rise and Fall is working "as a mod" so the game version numbers are the same with or without it.

**Comment by Azurency** [comment]
I can confirm that https://github.com/Azurency/CQUI_Community-Edition/releases/tag/mac-1.0.0.220 is working on Mac without rise and fall enabled.

**Comment by quiznilo1** [comment]
>Note for Mac/Linux users : as the Rise and Fall patch is still not avaible to your platform

Just a little quibble with 4194ca0ff444775adf695d7530d35dcbc0202358 merged 2 hours ago, Rise and Fall was released for Linux/Mac last Wednesday.

**Comment by wayneb64** [comment]
It was mentioned above there was a game update three days ago, but I cannot find any patch notes for that. I went into offline mode so I would not lose CQUI again. Can anyone confirm there was a patch to Vanilla and did it break the latest version of CQUI? What was in the latest update? Did they fix Nationalism?

Thanks

**Comment by sql4bucks** [comment]
@Azurency - Thanks so much for the mac-1.0.0.220 release, so nice to have CQUI again. I played the vanilla game for about 70 turns and it wasn't much fun.

**Comment by grgz** [comment]
@wayneb64 here are the patch notes for 1.0.0.216 and 1.0.0.220 when they were released on Windows last month.  
http://steamcommunity.com/games/289070/announcements/detail/1662261735494326654
http://steamcommunity.com/games/289070/announcements/detail/1662261735520460811

Civ6 on Mac/Linux has now received an update to 1.0.0.220.  When you update to this version you'll need to install the CQUI release for 1.0.0.220 that Azurency linked above.  Vanilla works fine with this version but R&F support is incomplete.

**Comment by evandy0** [comment]
New Civ6 player (downloaded from steam last weekend).  For some reason the mac-1.0.0.220 CQUI release wasn't working right for me... I couldn't move units, click&drag the map, or a couple of other things.   The menu dialog does say I have 1.0.0.220 (160865).

For what it's worth, I was able to make CQUI work for me by doing the following:
  1) Take the Patch-1.0.0.257 branch of CQUI
  2) replace the TechTree.lua, TechTree.xml, WorldTracker.lua, and WorldTracker.xml files from the base game and overwrite those in the CQUI folder

So I'm probably getting the base-game TechTree and WorldTracker, but at least the rest of CQUI is working right now.

**Comment by evandy0** [comment]
Heads up to all... I just booted CIV 6 last night, and the Mac version is now up to 1.0.0.260.  It /seems/ to work okay with the 1.0.0.257 branch.  We'll see what happens after I play more thoroughly.
